workspaces/global-header/plugins/global-header/src/components/SearchComponent/SearchBar.tsx

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright 2024 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from 'react';
+import {
+  SearchResultState,
+  SearchResultProps,
+} from '@backstage/plugin-search-react';
+import Typography from '@mui/material/Typography';
+import { Link } from '@backstage/core-components';
+import ListItem from '@mui/material/ListItem';
+import Box from '@mui/material/Box';
+import Divider from '@mui/material/Divider';
+import ArrowForwardIcon from '@mui/icons-material/ArrowForward';
+import Autocomplete from '@mui/material/Autocomplete';
+import TextField from '@mui/material/TextField';
+import InputAdornment from '@mui/material/InputAdornment';
+import SearchIcon from '@mui/icons-material/Search';
+import { highlightMatch } from '../../utils/stringUtils';
+import styles from './SearchBar.module.css';
+import { useNavigate } from 'react-router-dom';
+
+interface SearchBarProps {
+  query: SearchResultProps['query'];
+  setSearchTerm: (term: string) => void;
+}
+export const SearchBar = (props: SearchBarProps) => {
+  const { query, setSearchTerm } = props;
+  const navigate = useNavigate();
+
+  return (
+    <SearchResultState {...props}>
+      {({ loading, error, value }) => {
+        const results = query?.term ? value?.results ?? [] : [];
+        let options: string[] = [];
+        if (results.length > 0) {
+          options = [
+            ...results.slice(0, 5).map(result => result.document.title),
+            `${query?.term}`,
+          ];
+        } else if (query?.term) {
+          options = ['No results found'];
+        }
+
+        return (
+          <Autocomplete
+            freeSolo
+            options={options}
+            loading={loading}
+            getOptionLabel={option => option ?? ''}
+            onInputChange={(_, inputValue) => setSearchTerm(inputValue)}
+            sx={{ width: '100%' }}
+            filterOptions={x => x}
+            getOptionDisabled={option => option === 'No results found'}
+            onKeyDown={event => {
+              if (event.key === 'Enter') {
+                event.preventDefault();
+                if (query?.term) {
+                  navigate(`/search?query=${query?.term}`);
```

Comment:
We should always escape query paramters otherwise searches like `Tom & Jerry` will not work.

```suggestion
                  navigate(`/search?query=${encodeURIComponent(query?.term)}`);
```